packages/wrangler/src/pages/dev.ts

```diff
@@ -509,9 +638,13 @@ export const Handler = async (args: PagesDevArguments) => {
 				);
 				// Resolve anyway and run without Functions
 				onEnd();
-				// Turn off Functions
 				usingFunctions = false;
 			} else {
+				/*
+				 * do not start the `pages dev` session if we encounter errors
+				 * while attempting to build Functions. These flag underlying
+				 * issues in the Functions code, and should be addressed before
+				 */
```

Comment:
should we throw a `FatalError`?

Reply:
Similar to here, I think it makes sense. If a user has an error, it would be good to let them know as soon as possible: https://github.com/cloudflare/workers-sdk/pull/6150/files/b4006823e7ddeb4b5d8e656ba7f2f63756fe3602#diff-c90ff1ca6d929baaafb7d67d6d6ccd573593762adf4e7a9b6917343ceeceb68bR443

Reply:
✅